### ISSUE TITLE
For Database

### ISSUE THREAD

**Issue by praveenmalav09** [duste/FingerprintScannerRaspberryPi#6]
Can you please tell me how can i create and manage database for fingerprint Sensor R305 using raspberry pi 3?

**Comment by jcroucher** [comment]
The fingerprints are stored and matched on the scanner itself.
What you have to do is a multi-step process.

- create a new user in your local database with an unused ID in a range which matches the scanner from memory 0-254
- Initiate the registration process on the scanner sending this new ID
- On success mark the user as active in your local database or similar

then on scan
- initiate the scan process
- User places their finger on the scanner
- The scanner then returns an ID
- Lookup this ID in your local database to see who it is

**Comment by praveenmalav09** [comment]
but I want to use the sqlite3 database for large no. of entries so How can I use this? is it possible to use?

**Comment by jcroucher** [comment]
I don't believe it's possible with these style of fingerprint scanners. The scanning and fingerprint database is stored on the device itself.

**Comment by praveenmalav09** [comment]
I use Raspberry Pi 3 so which type of Sensor I use to use Database?

**Comment by praveenmalav09** [comment]
If I stored all template on the server and match with them. is it possible?

**Comment by duste** [comment]
You can pull the scanned fingerprint right off of the scanner.  if you have software that can process the image and compare the image or an algorithm to the sql server it can work.  you would be performing the image comparison on the RPi instead of letting the fingerprint scanner do that work for you.
